File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
from itertools import product
def solution(word):
    words = ['A','E','I','O','U']
    dic = []
    for i in range(1,6):
        dic.extend(list(product(words,repeat=i)))
    dic.sort()
    combined_dict = []
    
    for d in dic:
        combined_dict.append("".join(d))
    
    return combined_dict.index(word)+1

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    const vowels = ['A', 'E', 'I', 'O', 'U'];
    const counts = [781, 156, 31, 6, 1]; // 각 자리수별 가능한 단어 수
    
    let answer = 0;
    for (let i = 0; i < word.length; i++) {
        answer += vowels.indexOf(word[i]) * counts[i] + 1; // 해당 자리의 알파벳 인덱스에 따른 영향을 더해줌
    }
    
    return answer;
}